dynamiq/evaluations/metrics/bleu_score.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from functools import cached_property
+from typing import Callable
+
+from pydantic import BaseModel, ConfigDict, PrivateAttr, computed_field, model_validator
+
+logger = logging.getLogger(__name__)
+
+
+class RunInput(BaseModel):
+    """
+    Input model for the BLEU score evaluation.
+
+    Attributes:
+        ground_truth_answers (list[str]): List of reference strings, one per example.
+        answers (list[str]): List of response strings, one per example.
+    """
+
+    ground_truth_answers: list[str]
+    answers: list[str]
+
+    @model_validator(mode="after")
+    def check_equal_length(self) -> "RunInput":
+        """
+        Validate that the number of ground truth answers matches the number of responses.
+
+        Raises:
+            ValueError: If the lengths of `ground_truth_answers` and `answers` do not match.
+
+        Returns:
+            RunInput: The validated instance.
+        """
+        if len(self.ground_truth_answers) != len(self.answers):
+            raise ValueError("Answers and ground truth answers must have the same length.")
+        return self
+
+
+class RunOutput(BaseModel):
+    """
+    Output model for the BLEU score evaluation.
+
+    Attributes:
+        scores (list[float]): List of BLEU scores, one per reference/response pair.
+    """
+
+    scores: list[float]
+
+
+class BleuScoreEvaluator(BaseModel):
+    """
+    Evaluates BLEU scores using the sacrebleu library.
+
+    Attributes:
+        name (str): Name of the metric. Defaults to "BleuScore".
+    """
+
+    name: str = "BleuScore"
+
+    # Private attribute to store the sacrebleu corpus_bleu function
+    _corpus_bleu: Callable = PrivateAttr()
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @computed_field
+    @cached_property
+    def type(self) -> str:
```

Comment:
Let's maybe move this to some Base class of Mixin class to avoid copying this to each metric